pkg/plugins/datasource/wrapper/datasource_plugin_wrapper_v2.go

```diff
@@ -25,41 +25,42 @@ type DatasourcePluginWrapperV2 struct {
 	pluginType string
 }
 
-func (tw *DatasourcePluginWrapperV2) Query(ctx context.Context, ds *models.DataSource, query *tsdb.TsdbQuery) (*tsdb.Response, error) {
+func ModelToInstanceSettings(ds *models.DataSource) (*backend.DataSourceInstanceSettings, error) {
 	jsonDataBytes, err := ds.JsonData.MarshalJSON()
 	if err != nil {
 		return nil, err
 	}
 
+	return &backend.DataSourceInstanceSettings{
+		ID:                      ds.Id,
+		Name:                    ds.Name,
+		URL:                     ds.Url,
+		Database:                ds.Database,
+		User:                    ds.User,
+		BasicAuthEnabled:        ds.BasicAuth,
+		BasicAuthUser:           ds.BasicAuthUser,
+		JSONData:                jsonDataBytes,
+		DecryptedSecureJSONData: ds.DecryptedValues(),
+		Updated:                 ds.Updated,
+	}, nil
+}
+
+func (tw *DatasourcePluginWrapperV2) Query(ctx context.Context, ds *models.DataSource, query *tsdb.TsdbQuery) (*tsdb.Response, error) {
+	instanceSettings, err := ModelToInstanceSettings(ds)
+	if err != nil {
+		return nil, err
+	}
+
 	pbQuery := &pluginv2.QueryDataRequest{
-		Config: &pluginv2.PluginConfig{
-			OrgId:         ds.OrgId,
-			PluginId:      tw.pluginId,
-			LastUpdatedMS: ds.Updated.UnixNano() / int64(time.Millisecond),
-			DatasourceConfig: &pluginv2.DataSourceConfig{
-				Id:                      ds.Id,
-				Name:                    ds.Name,
-				Url:                     ds.Url,
-				Database:                ds.Database,
-				User:                    ds.User,
-				BasicAuthEnabled:        ds.BasicAuth,
-				BasicAuthUser:           ds.BasicAuthUser,
-				JsonData:                jsonDataBytes,
-				DecryptedSecureJsonData: ds.DecryptedValues(),
-			},
+		PluginContext: &pluginv2.PluginContext{
+			OrgId:                      ds.OrgId,
+			PluginId:                   tw.pluginId,
+			User:                       backend.ToProto().User(backendplugin.BackendUserFromSignedInUser(query.User)),
```

Comment:
This looks wrong. Would suggest populating User field in api package

Reply:
@marefr I think the Query call and the wrapper I need to keep the signature? If that is so I need to grab it from the tsdbQuery.